float64 cm_s
int64 way
bool walk_flag
leg_state L1
leg_state L2
leg_state L3
leg_state L4